src/test/java/HandleGetRequestTest.java

```diff
@@ -2,9 +2,9 @@
 import java.util.*;
 import java.io.IOException;
 
-public class HttpHandlerSelectorTest extends junit.framework.TestCase {
+public class HandleGetRequestTest extends junit.framework.TestCase {
 
-  HttpHandlerSelector handler;
+  HandleGetRequest getRequest;
```

Comment:
`ResponseFactory` sounds like a good name, `ResponseFactoryGetRequest` not so much. `GetRequestResponseFactory` probably makes more sense.

_However_, factory feels like the wrong abstraction here in general. This thing doesn't feel like it's a factory. It really feels more of a generic "request handler". It takes a request and gives you a response.

Factories are more about hiding the details of how to build or construct an object when that object can be built in different ways, or when we want to encapsulate the details of "how" to build it from the client of that object.

You would use a factory to build a service, or a "thing doer" of some sort that you later use to "do this thing". What's going on here is you're using a class, asking it to "do something or give me something" (based on the Request that's passed in) and what you're getting back is a data object that you _don't_ later use to do something on it, but rather retrieve some fields for presentation.

Perhaps, leave the names as they are for now. Maybe as you build up the next set of requirements a better name will come to our heads.